ykrt/src/mt.rs

```diff
@@ -244,6 +244,10 @@ impl MTThread {
                             Rc::get_mut(&mut self.inner).unwrap().tracer =
                                 Some((start_tracing(self.inner.tracing_kind), loc_id));
                             return None;
+                        } else {
+                            // We raced with another thread that's also trying to trace this
+                            // Location, so free the malloc'd block.
+                            unsafe { Box::from_raw(loc_id) };
```

Comment:
I wonder if we should just eagerly allocate each thread an ID upon it's creation. Then the ID can be reused across many tracing sessions at zero additional cost.

Reply:
Good idea. Once the current bug is fixed I'll look into that.

Reply:
And by extension, is the address of the `MTInner` a better ID?

Reply:
That feature *might* be disappearing at some point. Not